index.js

```diff
@@ -81,7 +81,7 @@ function _parseInput (input, opts, cb) {
 
   var commonPrefix = null
   input.forEach(function (item, i) {
-    if (typeof item === 'string') return
+    if (typeof item === 'string' || (Buffer.isBuffer(item) && !item.name)) return
```

Comment:
Also, this means that Buffers without name properties will sneak through. I don't see that they're ever assigned a name later.